raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -501,6 +506,10 @@ public void initialize(
         logger.info("Reading KRaft snapshot and log as part of the initialization");
         partitionState.updateState();
         logger.info("Starting voters are {}", partitionState.lastVoterSet());
+        if (nodeId.isPresent()) {
+            partitionState.setInitBootstrapNode(partitionState.lastVoterSet());
+            skipFirstAutoJoinAttempt = partitionState.initBootstrapNodes().contains(nodeId.getAsInt());
```

Comment:
If the `initBootstrapNodes` is only used in this check, do we still need to store the initBootstrapNode in `setInitBootstrapNode`? Can we do this directly?
`skipFirstAutoJoinAttempt = partitionState.lastVoterSet().contains(nodeId.getAsInt());`